base/abstractarray.jl

```diff
@@ -2221,6 +2241,33 @@ function _typed_hvncat(::Type{T}, ::Val{N}, as...) where {T, N}
     return A
 end
 
+
+# 0-dimensional cases for balanced and unbalanced hvncat methods
+
+_typed_hvncat(::Type{T}, ::Tuple{}, ::Bool, x) where T = fill(T(x))
+_typed_hvncat(::Type{T}, ::Tuple{}, ::Bool, x::Number) where T = fill(T(x))
+_typed_hvncat(::Type{T}, ::Tuple{}, ::Bool, x::AbstractArray) where T = T.(x)
+_typed_hvncat(::Type, ::Tuple{}, ::Bool, ::Number...) =
+    throw(ArgumentError("a 0-dimensional array may not have more than one element"))
+_typed_hvncat(::Type, ::Tuple{}, ::Bool, ::Any...) =
+    throw(ArgumentError("a 0-dimensional array may not have more than one element"))
+
+# balanced dimensions hvncat methods
+
+_typed_hvncat(T::Type, dims::Tuple{Int}, ::Bool, as...) = _typed_hvncat_1d(T, dims[1], Val(false), as...)
+_typed_hvncat(T::Type, dims::Tuple{Int}, ::Bool, as::Number...) = _typed_hvncat_1d(T, dims[1], Val(false), as...)
+
+function _typed_hvncat_1d(::Type{T}, ds::Int, ::Val{row_first}, as...) where {T, row_first}
+    lengthas = length(as)
+    ds > 0 ||
+        throw(ArgumentError("`dimsshape` argument must consist of positive integers"))
+    lengthas == ds ||
+        throw(ArgumentError("number of elements does not match `dimshape` argument; expected $ds, got $lengthas"))
+    return row_first ?
+        _typed_hvncat(T, Val(2), as...) :
+        _typed_hvncat(T, Val(1), as...)
```

Comment:
If it goes over multiple lines, I usually just use `if` blocks. Not really a big deal though.